src/types.rs

```diff
@@ -11,10 +13,41 @@ pub struct KeyValue {
 
 /// Represents a key-value pair that may be a tombstone.
 #[derive(Debug, Clone, PartialEq)]
-pub struct KeyValueDeletable {
+pub struct RowEntry {
     pub key: Bytes,
     pub value: ValueDeletable,
-    pub attributes: RowAttributes,
+    pub seq: u64,
+    pub flags: RowFlags,
+    pub create_ts: Option<i64>,
+    pub expire_ts: Option<i64>,
+}
+
+impl RowEntry {
+    #[allow(unused)]
+    pub fn new(
+        key: Bytes,
+        value: Option<Bytes>,
+        seq: u64,
+        create_ts: Option<i64>,
+        expire_ts: Option<i64>,
+    ) -> Self {
+        let flags = match &value {
+            Some(_) => RowFlags::default(),
+            None => RowFlags::Tombstone,
+        };
+        let value = match value {
+            Some(v) => ValueDeletable::Value(v),
+            None => ValueDeletable::Tombstone,
+        };
+        Self {
+            key,
+            value,
+            flags,
+            seq,
+            create_ts,
+            expire_ts,
+        }
+    }
 }
```

Comment:
Is `RowAttributes` still needed?

Reply:
currently `RowAttributes` is still referenced in MemTable, i'm planned to work on adding sequence number in `MemTable` in the next pr. i guess we can keep `RowAttributes` untouched here and have a further review about it after the refactor on Memtable finished.